### PR TITLE
fix: New chat aren't showing bold in Left Hand Nav

### PR DESCRIPTION
**Pull request by aswin-s**
### Details

This PR fixes an issue where new chat messages were not being shown in bold inside LHN.

### Fixed Issues

$ https://github.com/Expensify/App/issues/6874

### Tests

1.  Sign in as User A in one browser tab & as User B in another tab
2. Open chat with the other user in each tab.
3. Now navigate to a separate conversation in LHN for User B. (For example Concierge)
4. Send a message from User A to User B.
5. The new message should be highlighted in bold on the LHN for User B.
6. Verify that it works other way around as well.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

image

https://user-images.githubusercontent.com/6880914/147296695-7ecb1139-320b-4f91-8a5b-3b55dc6aa356.mp4

#### Mobile Web

N.A

#### Desktop

image

#### iOS

image

#### Android

Screenshot_1640299864

### CODE REVIEW

**Review by marcochavezf — APPROVED** [review]
Looks good to me! cc @parasharrajat

File: src/Expensify.js [hunk 1]
```diff
@@ -134,7 +135,8 @@ class Expensify extends PureComponent {
     }
 
     componentWillUnmount() {
-        AppState.removeEventListener('change', this.initializeClient);
+        if (!this.appStateChangeListener) { return; }
```
Comment (parasharrajat) [inline src/Expensify.js:138]: Could you please expand this to multiple lines?
```suggestion
        if (!this.appStateChangeListener) {
        	return;
        }
```

File: src/Expensify.js [hunk 2]
```diff
@@ -85,6 +85,7 @@ class Expensify extends PureComponent {
         ActiveClientManager.init();
         this.setNavigationReady = this.setNavigationReady.bind(this);
         this.initializeClient = this.initializeClient.bind(true);
+        this.appStateChangeListener = null;
```
Comment (parasharrajat) [inline src/Expensify.js:88]:
```suggestion

```

File: src/pages/home/report/ReportActionsView.js [hunk 3]
```diff
@@ -106,6 +106,7 @@ class ReportActionsView extends React.Component {
         this.recordTimeToMeasureItemLayout = this.recordTimeToMeasureItemLayout.bind(this);
         this.loadMoreChats = this.loadMoreChats.bind(this);
         this.sortedReportActions = [];
+        this.appStateChangeListener = null;
```
Comment (parasharrajat) [inline src/pages/home/report/ReportActionsView.js:109]:
```suggestion

```

**Comment by marcochavezf** [comment]
Merging it and applying the `CP Staging` label because more employees are going to experience this issue.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Thanks for the changes @aswin-s and waiting for review I was on vacation. This looks good. 

Could you please add videos for all platforms and present the following features;
1. The current issue.
2. Unread Message when the app is put to foreground from background.

**Comment by aswin-s** [comment]
@parasharrajat Should I raise a follow up PR with the suggested changes?

**Comment by parasharrajat** [comment]
These changes are no blocker for me so If @marcochavezf says we should do these then yeah please add a follow-up?

**Comment by marcochavezf** [comment]
> These changes are no blocker for me so If @marcochavezf says we should do these then yeah please add a follow-up?

@parasharrajat Yeah, I think we could include these changes as a follow-up, thank you for doing it @aswin-s

**Comment by roryabraham** [comment]
Hi, I'm now seeing the following error on `main`:

image

**Comment by roryabraham** [comment]
It looks like this change was made in react-native `0.65`, but we're still using react-native `0.64.1`

**Comment by roryabraham** [comment]
I think we need to either revert this PR or upgrade the whole app to RN `0.65`

**Comment by aswin-s** [comment]
@roryabraham as explained over [here](https://github.com/Expensify/App/issues/6874#issuecomment-1000493831), removeEventListener was not working on react-native-web. I guess upgrading to  0.65 will solve it for both platforms, considering there is only a small diff between both versions. 

https://react-native-community.github.io/upgrade-helper/?from=0.64.1&to=0.65.0

Related to this, I would also like to point out that dependencies like `react-native-modal` is also consistently throwing similar error on web which is fixed in their latest release. Unfortunately that dependency upgrade is also blocked by react-native peer dependency requirement of ">=0.65.0"

WDYT?

**Comment by roryabraham** [comment]
I suppose we should update react-native and the other dependencies that are throwing the error.

**Comment by roryabraham** [comment]
I have an upgrade PR [here](https://github.com/Expensify/App/pull/6914), but I could use help testing it. Works well on iOS from my testing so far.

**Pull request merged**